interpreter/syntax/ast.ml

```diff
@@ -100,7 +100,7 @@ and instr' =
   | Suspend of idx                    (* suspend continuation *)
   | Resume of (idx * idx) list        (* resume continuation *)
   | ResumeThrow of idx                (* abort continuation *)
-  | Guard of block_type * instr list  (* guard against suspension *)
+  | Barrier of block_type * instr list  (* guard against suspension *)
```

Comment:
Barriers, are they similar in spirit to Racket's [continuation barriers](https://docs.racket-lang.org/reference/eval-model.html#%28tech._continuation._barrier%29)?